### PR TITLE
Made the site a PWA

### PR DESCRIPTION
**Pull request by klezi10**
|                                    Web Dev Path                                    |
| :--------------------------------------------------------------------------------: |
| [113](https://github.com/MarianaSouza/web-dev-path/issues/113) |

#### Have you updated the CHANGELOG.md file? If not, please do it.
Yes.

#### What is this change?
Made the site a Progressive Web App (PWA)

image

#### Were there any complications while making this change?
Yes, I have been unable to test it with the correct link which is why the PR is still a work in progress. I am currently working on this.

#### How did you verify this change?
Locally. It passed the Lighthouse Progressive Web App test with the localhost link. 

#### When should this be merged?
Once it has been tested in production and after review.

### CODE REVIEW

File: public/favicon/site.webmanifest [hunk 1]
```diff
@@ -1,19 +1,27 @@
 {
     "name": "Web Dev Path",
     "short_name": "Web Dev Path",
+    "start_url": "https://webdevpath.co/",
```
Comment (cherylli) [inline public/favicon/site.webmanifest:4]: `"start_url": "/",`
Changing to this works even in development mode.

**Review by cherylli — COMMENTED** [review]
image

**Review by cherylli — APPROVED** [review]

**Review by mariana-caldas — CHANGES_REQUESTED** [review]
This looks great! We're almost there, @klezi10 and @cherylli !

As explained in the end of [this video](https://www.youtube.com/watch?v=ARNN_zmrwcw&t=714s), we shouldn't commit the auto-generated files sw.js** and worboxes into production.

Could you please remove those 4 files, add the following code to `.gitignore` , generate them locally again by running `yarn install`, test everything locally again and commit the changes to the branch making sure those files are properly ignored? Thanks!
```
# PWA files
**/public/sw.js
**/public/sw.js.map
**/public/worker-*.js
**/public/workbox-*.js
**/public/worker-*.js
**/public/workbox-*.js.map
**/public/worker-*.js.map

**Review by mariana-caldas — APPROVED** [review]
This looks great, @klezi10 !
Could you please wait until Monday to merge it so we give others a chance to review it too?
Thanks so much!

**Comment by vmcodes** [comment]
Looks great to me. Double checked the Lighthouse reports and saw no issues. 

An off topic suggestion I noticed was that we we're missing a canonical link and robots.txt file which would help with Google indexing :smile:

**Pull request merged**